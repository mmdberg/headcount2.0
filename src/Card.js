import React, { Component } from 'react';
import './styles/card.css';
import PropTypes from 'prop-types';

class Card extends Component {  
  cleanYears = (stats) => {
    return Object.keys(stats).map(year => {
      let roundedPerc = parseFloat(stats[year].toFixed(2));
      if (roundedPerc < 0.5) {
        return <p key={year}>{year}: {roundedPerc}</p>;
      } else {
        return <p key={year} className="goodSchool">{year}: {roundedPerc}</p>;
      }       
    });
  }

  handleClick = () => {
    if (this.props.className.includes('clicked')) {
      this.props.deselectCard(this.props);
    } else {
      this.props.selectCard(this.props);
    }
  }

  render() {
    return (
      <div className={this.props.className} onClick={this.handleClick}>
        <h2>{this.props.location}</h2>
        {this.cleanYears(this.props.stats)}
      </div>
    );
  }
}


Card.propTypes = {
  stats: PropTypes.object.isRequired,
  location: PropTypes.string.isRequired,
  selectCard: PropTypes.func,
  deselectCard: PropTypes.func,
  className: PropTypes.string
};

export default Card;