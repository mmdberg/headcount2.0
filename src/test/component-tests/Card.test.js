import React from 'react';
import { shallow, mount } from 'enzyme';
import mockCardData from '../../data/mock_card_data'
import Card from '../../Card'


describe('Card', () => {

  let wrapper;

  const mockFunction = jest.fn();

  beforeEach( () => {
    wrapper = shallow(<Card data={mockCardData} />)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should add class goodSchool to values over .5', () => {
    wrapper.instance().yearValues(mockCardData);
    expect(wrapper.find('.goodSchool').length).toEqual(7)
  })

  it('should call selectCard on click if not selected',() => {
    wrapper = shallow(<Card className='card' data={mockCardData} selectCard={mockFunction}/>)
    wrapper.instance().handleClick()
    expect(mockFunction).toHaveBeenCalled()
  })

  it('should call deselectCard on click if selected', () => {
    wrapper = shallow(<Card className='card clicked' data={mockCardData} deselectCard={mockFunction}/>)
    wrapper.instance().handleClick();
    expect(mockFunction).toHaveBeenCalled();
  })
})