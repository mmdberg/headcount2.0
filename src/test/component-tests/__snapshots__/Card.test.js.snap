// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "2004": 0.24,
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
            "2008": 0.536,
            "2009": 0.598,
            "2010": 0.64,
            "2011": 0.672,
            "2012": 0.695,
            "2013": 0.703,
            "2014": 0.741,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        Array [
          <p>
            2004
            : 
            0.24
</p>,
          <p>
            2005
            : 
            0.28
</p>,
          <p>
            2006
            : 
            0.34
</p>,
          <p>
            2007
            : 
            0.4
</p>,
          <p
            className="goodSchool"
>
            2008
            : 
            0.54
</p>,
          <p
            className="goodSchool"
>
            2009
            : 
            0.6
</p>,
          <p
            className="goodSchool"
>
            2010
            : 
            0.64
</p>,
          <p
            className="goodSchool"
>
            2011
            : 
            0.67
</p>,
          <p
            className="goodSchool"
>
            2012
            : 
            0.69
</p>,
          <p
            className="goodSchool"
>
            2013
            : 
            0.7
</p>,
          <p
            className="goodSchool"
>
            2014
            : 
            0.74
</p>,
        ],
      ],
      "className": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "2004",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2004",
            ": ",
            0.24,
          ],
        },
        "ref": null,
        "rendered": Array [
          "2004",
          ": ",
          0.24,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2005",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2005",
            ": ",
            0.28,
          ],
        },
        "ref": null,
        "rendered": Array [
          "2005",
          ": ",
          0.28,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2006",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2006",
            ": ",
            0.34,
          ],
        },
        "ref": null,
        "rendered": Array [
          "2006",
          ": ",
          0.34,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2007",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2007",
            ": ",
            0.4,
          ],
        },
        "ref": null,
        "rendered": Array [
          "2007",
          ": ",
          0.4,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2008",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2008",
            ": ",
            0.54,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2008",
          ": ",
          0.54,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2009",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2009",
            ": ",
            0.6,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2009",
          ": ",
          0.6,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2010",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2010",
            ": ",
            0.64,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2010",
          ": ",
          0.64,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2011",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2011",
            ": ",
            0.67,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2011",
          ": ",
          0.67,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2012",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2012",
            ": ",
            0.69,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2012",
          ": ",
          0.69,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2013",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2013",
            ": ",
            0.7,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2013",
          ": ",
          0.7,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2014",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2014",
            ": ",
            0.74,
          ],
          "className": "goodSchool",
        },
        "ref": null,
        "rendered": Array [
          "2014",
          ": ",
          0.74,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          Array [
            <p>
              2004
              : 
              0.24
</p>,
            <p>
              2005
              : 
              0.28
</p>,
            <p>
              2006
              : 
              0.34
</p>,
            <p>
              2007
              : 
              0.4
</p>,
            <p
              className="goodSchool"
>
              2008
              : 
              0.54
</p>,
            <p
              className="goodSchool"
>
              2009
              : 
              0.6
</p>,
            <p
              className="goodSchool"
>
              2010
              : 
              0.64
</p>,
            <p
              className="goodSchool"
>
              2011
              : 
              0.67
</p>,
            <p
              className="goodSchool"
>
              2012
              : 
              0.69
</p>,
            <p
              className="goodSchool"
>
              2013
              : 
              0.7
</p>,
            <p
              className="goodSchool"
>
              2014
              : 
              0.74
</p>,
          ],
        ],
        "className": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "2004",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2004",
              ": ",
              0.24,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2004",
            ": ",
            0.24,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2005",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2005",
              ": ",
              0.28,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2005",
            ": ",
            0.28,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2006",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2006",
              ": ",
              0.34,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2006",
            ": ",
            0.34,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2007",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2007",
              ": ",
              0.4,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2007",
            ": ",
            0.4,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2008",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2008",
              ": ",
              0.54,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2008",
            ": ",
            0.54,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2009",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2009",
              ": ",
              0.6,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2009",
            ": ",
            0.6,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2010",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2010",
              ": ",
              0.64,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2010",
            ": ",
            0.64,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2011",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2011",
              ": ",
              0.67,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2011",
            ": ",
            0.67,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2012",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2012",
              ": ",
              0.69,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2012",
            ": ",
            0.69,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2013",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2013",
              ": ",
              0.7,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2013",
            ": ",
            0.7,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2014",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2014",
              ": ",
              0.74,
            ],
            "className": "goodSchool",
          },
          "ref": null,
          "rendered": Array [
            "2014",
            ": ",
            0.74,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
